src/scripts/generate-readme.ts

```diff
@@ -0,0 +1,106 @@
+import fs from 'node:fs'
+
+interface GetUsageSectionOptions<TNullable = false> {
+  isNullable?: TNullable
+}
+
+const getUsageSection = <TNullable = false>(
+  readmePath: string,
+  options: GetUsageSectionOptions<TNullable> = {},
+): TNullable extends true ? string | null : string => {
+  const { isNullable = false } = options
+
+  try {
+    return readUsageSection(readmePath)
+  } catch (error) {
+    if (isNullable === true) {
+      return null as any
+    }
+
+    throw error
+  }
+}
+
+const readUsageSection = (readmePath: string): string => {
+  const data = fs.readFileSync(readmePath, {
+    encoding: 'utf-8',
+  })
+
+  const regex = /\s\S*#* Usage\s*([^#]*)/
+  const matches = regex.exec(data)
+  if (matches == null || matches.length !== 2) {
+    throw new Error('Missing [## Usage] section')
+  }
+
+  const usage = matches[1]?.replace(/^\s+|\s+$/g, '') // trim any trailing whitespace
+
+  if (usage == null) {
+    throw new Error('Invalid [## Usage] format')
+  }
+
+  return usage
+}
+
+void (async () => {
```

Comment:
This is non-blocking, but more a suggestion for improved code readability organization. 

Types and logic might be a little simpler (and you can re-use some of the logic) if you break it up like this.

Instead of a self exec function, it can be done like here: https://github.com/seamapi/javascript-http/blob/main/generate-routes.ts#L554-L556

```ts
// just leave out @seamapi/types and assume missing usage section is always an error
// since we will just add it here if we want to include it later.
// Should greatly simplify all the null conditionals
const submodules = [
  path.join('@seamapi', 'webhook'),
  path.join('@seamapi', 'http'),
]

const writeReadmeUsageSection = async (content: string): Promise<void> => {}
const readUsageSection = async (content: string): string => {}

const sections = await Promise.all(submodules.map(readUsageSection))
await writeReadmeUsage(sections.join('\n')
```